gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -236,7 +237,7 @@ protected void processMessage(DecodeableKafkaRecord<byte[],byte[]> message) {
               try {
                 this.dagManagementStateStore.addJobDagAction(flowGroup, flowName, flowExecutionId, jobName, DagActionStore.DagActionType.REEVALUATE);
               } catch (IOException e) {
-                if (e.getCause() != null && isThrowableInstanceOf(e.getCause(), nonRetryableExceptions)) {
+                if (ExceptionUtils.isExceptionInstanceOf(e, nonRetryableExceptions)) {
```

Comment:
here we only create duplicate REEVALUATE for transient so it should be `!ExceptionUtils.isExceptionInstanceOf` then we create another action. Can you also add a metric here to `DagProcessingEngineMetrics`?

Reply:
yes, we only re try adding dag action for transient exceptions, so we want it to throw exception, so it is retried. if it is not retryable, it should be swallowed. so the condition looks correct to me...